### PR TITLE
feat(ui): add refresh and autorefresh buttons in events view page

### PR DESCRIPTION
**Pull request by kduret**
## Description

image

* refresh interval is got from administration > parameters > monitoring
* default refresh interval is 15s
* each refresh, search, or sort reset the refresh interval

**Fixes** MON-5093

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

* Check that refresh button refresh the listing

* Stop autorefresh and check that listing is not refresh

### CODE REVIEW

File: www/front_src/src/Resources/index.tsx [hunk 1]
```diff
@@ -94,6 +96,12 @@ const Resources = (): JSX.Element => {
   >(null);
 
   const [loading, setLoading] = useState(true);
+  const [enabledAutorefresh, setEnabledAutorefresh] = useState(true);
+
+  const refreshIntervalMs = useSelector(
+    (state) => (state?.intervals?.AjaxTimeReloadMonitoring || 15) * 1000,
```
Comment (bdauria) [inline www/front_src/src/Resources/index.tsx:102]: Isn't it better to use a default value in the store directly?

File: www/front_src/src/Resources/index.tsx [hunk 2]
```diff
@@ -129,14 +137,30 @@ const Resources = (): JSX.Element => {
       .finally(() => setLoading(false));
   };
 
+  const initAutorefresh = (): void => {
+    window.clearInterval(refreshIntervalRef.current);
```
Comment (bdauria) [inline www/front_src/src/Resources/index.tsx:141]:
```suggestion
    clearInterval(refreshIntervalRef.current);
```

File: www/front_src/src/Resources/index.tsx [hunk 3]
```diff
@@ -129,14 +137,30 @@ const Resources = (): JSX.Element => {
...
+
+    const interval = enabledAutorefresh
+      ? window.setInterval(load, refreshIntervalMs)
```
Comment (bdauria) [inline www/front_src/src/Resources/index.tsx:144]:
```suggestion
      ? setInterval(load, refreshIntervalMs)
```

File: www/front_src/src/Resources/index.tsx [hunk 4]
```diff
@@ -129,14 +137,30 @@ const Resources = (): JSX.Element => {
...
+
+  const initAutorefreshAndLoad = (): void => {
+    initAutorefresh();
+    load();
+  };
+
   useEffect(() => {
     return (): void => {
       tokenSource.cancel();
+      window.clearInterval(refreshIntervalRef.current);
```
Comment (bdauria) [inline www/front_src/src/Resources/index.tsx:158]:
```suggestion
      clearInterval(refreshIntervalRef.current);
```

File: www/front_src/src/Resources/index.tsx (hunk 3 above)
Comment (kduret) [reply www/front_src/src/Resources/index.tsx:144]: typescript does not work with this change

File: www/front_src/src/Resources/index.tsx (hunk 2 above)
Comment (kduret) [reply www/front_src/src/Resources/index.tsx:141]: typescript does not work with this change

File: www/front_src/src/Resources/index.tsx (hunk 4 above)
Comment (kduret) [reply www/front_src/src/Resources/index.tsx:158]: typescript does not work with this change

File: www/front_src/src/Resources/index.tsx (hunk 1 above)
Comment (kduret) [reply www/front_src/src/Resources/index.tsx]: done

File: www/front_src/src/Resources/index.tsx (hunk 2 above)
Comment (kduret) [reply www/front_src/src/Resources/index.tsx:141]: https://github.com/microsoft/TypeScript/issues/842#issuecomment-339860768

**Pull request merged**